demos/node-classification-gat/gat-cora-example.py

```diff
@@ -0,0 +1,360 @@
+    )
+    print(model.summary())
+
+    # Train model
+    # Callbacks
+    if not os.path.isdir("logs"):
+        os.makedirs("logs")
+    N = len(node_ids)
+    es_callback = EarlyStopping(monitor="val_weighted_acc", patience=es_patience)
+    tb_callback = TensorBoard(batch_size=N)
+    mc_callback = ModelCheckpoint(
+        "logs/best_model.h5",
+        monitor="val_weighted_acc",
+        save_best_only=True,
+        save_weights_only=True,
+    )
+
+    if args.interface == "fit":
+        print("\nUsing model.fit() to train the model\n")
+        # Get the training data
+        [X, A], y_train, node_mask_train = train_gen.__getitem__(0)
+        N = A.shape[0]
+        # A = sparse.csr_matrix(A + np.eye(A.shape[0]))  # Add self-loops
+
+        # Get the validation data
+        [_, _], y_val, node_mask_val = val_gen.__getitem__(0)
+
+        history = model.fit(
+            x=[X, A],
+            y=y_train,
+            sample_weight=node_mask_train,
+            batch_size=N,
+            shuffle=False,  # must be False, since shuffling data means shuffling the whole graph
+            epochs=num_epochs,
+            verbose=2,
+            validation_data=([X, A], y_val, node_mask_val),
+            callbacks=[es_callback, tb_callback, mc_callback],
+        )
+    else:
+        print("\nUsing model.fit_generator() to train the model\n")
+        history = model.fit_generator(
+            train_gen,
+            epochs=num_epochs,
+            validation_data=val_gen,
+            verbose=2,
+            shuffle=False,
+            callbacks=[es_callback, tb_callback, mc_callback],
+        )
+
+    # Load best model
+    model.load_weights("logs/best_model.h5")
+
+    # Evaluate on validation set and print metrics
+    if args.interface == "fit":
+        val_metrics = model.evaluate(
+            x=[X, A], y=y_val, sample_weight=node_mask_val, batch_size=N
+        )
+    else:
+        val_metrics = model.evaluate_generator(val_gen)
+
+    print("\nBest model's Validation Set Metrics:")
+    for name, val in zip(model.metrics_names, val_metrics):
+        print("\t{}: {:0.4f}".format(name, val))
+
+    # Evaluate on test set and print metrics
+    if args.interface == "fit":
+        [_, _], y_test, node_mask_test = generator.flow(
+            test_nodes, test_targets
+        ).__getitem__(0)
+        test_metrics = model.evaluate(
+            x=[X, A], y=y_test, sample_weight=node_mask_test, batch_size=N
+        )
+    else:
+        test_metrics = model.evaluate_generator(
+            generator.flow(test_nodes, test_targets)
+        )
+
+    print("\nBest model's Test Set Metrics:")
+    for name, val in zip(model.metrics_names, test_metrics):
+        print("\t{}: {:0.4f}".format(name, val))
+
+    # Get predictions for all nodes
+    if args.interface == "fit":
+        all_predictions = model.predict(x=[X, A], batch_size=N)
+    else:
+        all_predictions = model.predict_generator(generator.flow(node_ids))
```

Comment:
The `predict` or `predict_generator` function now operates differently to the `GraphSAGE` model in that if you give it less than the complete set of nodes, it will still return all predictions and in a fixed order.

This is confusing for users – whatever is given inthe `flow` argument will be ignored by the `predict_generator` as it's just used for masking.

Reply:
I've added a comment about this in both the demos